From: =?utf-8?q?Florian_M=C3=BCllner?= <fmuellner@gnome.org>
Date: Sun, 12 Apr 2020 17:15:48 +0200
Subject: appDisplay: Look up directory- instead of category translations

Translations are provided by .directory files, so trying to look
up a category name without the suffix will always fail.

Commit 343b3351f1 tried to fix this previously by changing the
saved keys, but that broke existing translatable folders.

Appending the .directory suffix for the lookup instead fixes the
issue without regressing non-custom folders.

https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/2623

Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/gnome-shell/+bug/1872434
Origin: https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1196
---
 js/ui/appDisplay.js | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/js/ui/appDisplay.js b/js/ui/appDisplay.js
index 991ab2a..2bc7d93 100644
--- a/js/ui/appDisplay.js
+++ b/js/ui/appDisplay.js
@@ -114,7 +114,8 @@ function _findBestFolderName(apps) {
     }, commonCategories);
 
     for (let category of commonCategories) {
-        let translated = Shell.util_get_translated_folder_name(category);
+        const directory = '%s.directory'.format(category);
+        const translated = Shell.util_get_translated_folder_name(directory);
         if (translated !== null)
             return translated;
     }
